### DISCUSSION TITLE
Alignment for elements on a single line

### DISCUSSION THREAD

**Discussion by CasparNuel** [typst/typst#2647] [Q&A] [answered]
I would like to be able to put two elements on the same line, but with different alignment. I have looked at the option to use `#grid` but this is not really what I want so far. Perhaps I am using it wrong.

For example, let's say I want the word "left" completely at the left of a line, and the word "right" completely to the right of a line. I tried this:

```typ
#align(left)[left] #align(right)[right]
```

This results in:

![image](https://github.com/typst/typst/assets/73889285/b2509bbc-abc1-4c1a-9a1b-296ae56153d3)

I want the text on the same line.

My next Idea was to just type the text and use something like `#v(fill)` but this is not currently a thing in Typst.

Can anyone help me achieve the alignment of two pieces of text on a single line? Thanks!

**Comment by laurmaedje** [comment, marked as answer]
Try `#h(1fr)`

**Reply by CasparNuel** [reply]
I tried
```typ
#align(left)[left] #h(1fr) #align(right)[right]

```
But this yielded the same result. Is this what you meant or did I understand it wrong?

**Reply by CasparNuel** [reply]
Never mind lol, dropping the align obviously did the trick! Thanks.
